lib/widgets/katex.dart

```diff
@@ -0,0 +1,202 @@
+          padding.vertical;
+    }
+    return padding.vertical;
+  }
+
+  @override
+  double computeMaxIntrinsicHeight(double width) {
+    final EdgeInsets padding = _resolvedPadding;
+    if (child != null) {
+      // Relies on double.infinity absorption.
+      return child!.getMaxIntrinsicHeight(math.max(0.0, width - padding.horizontal)) +
+          padding.vertical;
+    }
+    return padding.vertical;
+  }
+
+  @override
+  @protected
+  Size computeDryLayout(covariant BoxConstraints constraints) {
+    final EdgeInsets padding = _resolvedPadding;
+    if (child == null) {
+      return constraints.constrain(Size(padding.horizontal, padding.vertical));
+    }
+    final BoxConstraints innerConstraints = constraints.deflate(padding);
+    final Size childSize = child!.getDryLayout(innerConstraints);
+    return constraints.constrain(
+      Size(padding.horizontal + childSize.width, padding.vertical + childSize.height),
+    );
+  }
+
+  @override
+  double? computeDryBaseline(covariant BoxConstraints constraints, TextBaseline baseline) {
+    final RenderBox? child = this.child;
+    if (child == null) {
+      return null;
+    }
+    final EdgeInsets padding = _resolvedPadding;
+    final BoxConstraints innerConstraints = constraints.deflate(padding);
+    final BaselineOffset result =
+        BaselineOffset(child.getDryBaseline(innerConstraints, baseline)) + padding.top;
+    return result.offset;
+  }
+
+  @override
+  void performLayout() {
+    final BoxConstraints constraints = this.constraints;
+    final EdgeInsets padding = _resolvedPadding;
+    if (child == null) {
+      size = constraints.constrain(Size(padding.horizontal, padding.vertical));
+      return;
+    }
+    final BoxConstraints innerConstraints = constraints.deflate(padding);
+    child!.layout(innerConstraints, parentUsesSize: true);
+    final BoxParentData childParentData = child!.parentData! as BoxParentData;
+    childParentData.offset = Offset(padding.left, padding.top);
+    size = constraints.constrain(
+      Size(padding.horizontal + child!.size.width, padding.vertical + child!.size.height),
+    );
+  }
+
+  @override
+  void debugPaintSize(PaintingContext context, Offset offset) {
+    super.debugPaintSize(context, offset);
+    assert(() {
+      final Rect outerRect = offset & size;
+      debugPaintPadding(
+        context.canvas,
+        outerRect,
+        child != null ? _resolvedPaddingCache!.deflateRect(outerRect) : null,
+      );
+      return true;
+    }());
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DiagnosticsProperty<EdgeInsetsGeometry>('padding', padding));
+    properties.add(EnumProperty<TextDirection>('textDirection', textDirection, defaultValue: null));
+  }
+}
+
+extension on EdgeInsetsGeometry {
+  bool get isNegative => !isNonNegative;
```

Comment:
I think this is clearer (and hardly much longer) if just inlined.

In particular there's some ambiguity in that "is negative" sounds like it might require *all* the sides to be negative. So being more transparent, with one fewer layer of indirection, helps mitigate that.